scripts/src/primazactl/primazamain/parser.py

```diff
@@ -0,0 +1,112 @@
+import os
+import argparse
+from pathlib import Path
+from primazactl.errors import AtLeastOneError
+from primazactl.types import \
+    existing_file, kubernetes_name, semvertag_or_latest
+from .primazamain import PrimazaMain
+from primazactl.utils.kubeconfig import from_env
+
+
+def add_group(subparsers, parents=[]):
+    main_group = subparsers.add_parser(
+        name="main",
+        help="Operations on main Primaza cluster",
+        parents=parents)
+    main_subparsers = main_group.add_subparsers()
+
+    add_subparser_install(main_subparsers, parents=parents)
+    add_subparser_uninstall(main_subparsers, parents=parents)
+
+
+def add_subparser_install(parser: argparse.ArgumentParser, parents=[]):
+    install_parser = parser.add_parser(
+        name="install",
+        help="Install Primaza on target cluster",
+        parents=parents)
+    install_parser.set_defaults(func=install_primaza)
+    add_shared_args(install_parser)
+
+
+def add_subparser_uninstall(parser: argparse.ArgumentParser, parents=[]):
+    uninstall_parser = parser.add_parser(
+        "uninstall",
+        help="Uninstall Primaza on target cluster",
+        parents=parents)
+    uninstall_parser.set_defaults(func=uninstall_primaza)
+    add_shared_args(uninstall_parser)
+
+
+def add_shared_args(parser: argparse.ArgumentParser):
+    # Primaza
+    parser.add_argument(
+        "-f", "--config",
+        dest="config",
+        required=False,
+        help="primaza config file. Takes precedence over --version",
+        type=existing_file)
```

Comment:
This argument and several others are now defined in two places, would rather see them defined once only.